rig/place_and_route/utils.py

```diff
@@ -360,6 +364,54 @@ def build_routing_tables(routes, net_keys, omit_default_routes=True):
     return routing_tables
 
 
+def build_and_minimise_routing_tables(routes, net_keys, target_length=1023):
+    """Convert a set of RoutingTrees into a per-chip set of routing tables and
+    attempt to minimise those tables to meet a target length.
+
+    This is a wrapper around :py:func:`~.build_routing_tables` and
+    :py:func:`rig.routing_table.minimise`.
+
+    .. warning::
+        It may not be possible to minimise all tables such that they are
+        shorter than `target_length`. The chip co-ordinates of any routing
+        tables that could not be sufficiently minimised are reported as the
+        second returned value.
```

Comment:
I'd imagine that any application which doesn't meet the target given would have no option at this point but to bail out. As such I'd suggest just making this throw an exception instead.